experiments/AMIP/modular/components/atmosphere/climaatmos_init.jl

```diff
@@ -21,10 +21,32 @@ struct ClimaAtmosSimulation{P, Y, D, I} <: AtmosModelSimulation
 end
 name(::ClimaAtmosSimulation) = "ClimaAtmosSimulation"
 
-function atmos_init(::Type{FT}, parsed_args::Dict) where {FT}
+"""
+    get_atmos_config(coupler_dict::Dict)
+
+Returns the specified atmospheric configuration (`atmos_config_dict`) overwitten by arguments
+in the coupler dictionary (`config_dict`).
+"""
+function get_atmos_config(coupler_dict)
+    atmos_file = coupler_dict["atmos_config_file"]
+    # override default or specified configs with coupler arguments, and set the correct atmos config_file
+    if isnothing(atmos_file)
+        @info "Using Atmos default configuration"
+        merge(CA.override_default_config(atmos_file), coupler_dict, Dict("config_file" => atmos_file))
+    else
+        @info "Using Atmos configuration from $atmos_file"
+        merge(
+            CA.override_default_config(joinpath(pkgdir(CA), atmos_file)),
```

Comment:
Why do we need to call `joinpath` here but not in the `if` branch?

Reply:
In the if part, `atmos_file = nothing`, so this [override function](https://github.com/CliMA/ClimaAtmos.jl/blob/e2268a7b3c9920df5afa10484672f13f5a6e6d81/src/solver/yaml_helper.jl#L63) defaults to `CA.default_config_dict()`. But you're right that this is not obvious. I'll chage this to `CA.default_config_dict()`.